cmake/modules/Buildpmem.cmake

```diff
@@ -48,6 +49,9 @@ function(build_pmem)
   set_target_properties(pmem::pmem PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES ${PMDK_INCLUDE}
     IMPORTED_LOCATION "${PMDK_LIB}/libpmem.a"
+    IMPORTED_LOCATION "${PMDK_LIB}/libpmem2.a"
+    IMPORTED_LOCATION "${PMDK_LIB}/libpmemcommon.a"
+    IMPORTED_LOCATION "${PMDK_LIB}/libpmemcore.a"
```

Comment:
@optimistyzy  this fails the build. 

```
[100%] Linking CXX executable ../../../bin/unittest_librbd
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_memmove'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_deep_flush'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_msync'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_persist'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_flush'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_memset'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_drain'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_is_pmem'
../../../lib/libceph_librbd_pwl_cache.so.1.0.0: undefined reference to `pmem_memcpy'
collect2: error: ld returned 1 exit status
make[2]: *** [src/test/librbd/CMakeFiles/unittest_librbd.dir/build.make:1652: bin/unittest_librbd] Error 1
make[1]: *** [CMakeFiles/Makefile2:16137: src/test/librbd/CMakeFiles/unittest_librbd.dir/all] Error 2
```

see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/62138//consoleFull

it seems that we can have only a single `IMPORTED_LOCATION` for each target. see https://cmake.org/cmake/help/latest/prop_tgt/IMPORTED_LOCATION.html.

an alternative is to have multiple targets for each of these pmdk libraries, something like
```cmake
foreach(c pmem pmem2 pmemcommon pmemcore pmemobj)
  add_library(pmem::$c STATIC IMPORTED GLOBAL)
  add_dependencies(pmem::$c pmdk_ext)
  set_target_properties(pmem::$c PROPERTIES
    INTERFACE_INCLUDE_DIRECTORIES ${PMDK_INCLUDE}
    IMPORTED_LOCATION "${PMDK_LIB}/lib${c}.a")
endforeach()
set_property(TARGET pmem::pmemobj
  PROPERTY INTERFACE_LINK_LIBRARIES "pmem::pmem;${CMAKE_THREAD_LIBS_INIT}")
# add more inter-library dependencies here
```

and (only) link against the used pmem libraries.

Reply:
This is a local test version, which should not be included in this PR.

Reply:
fair enough. will retest this PR.